src/schemes/fluid/weakly_compressible_sph/state_equations.jl

```diff
@@ -49,3 +49,65 @@ function inverse_state_equation(state_equation::StateEquationCole, pressure)
 
     return reference_density * tmp^(1 / exponent)
 end
+
+@inline sound_speed(eos) = eos.sound_speed
+
+@doc raw"""
+    StateEquationIdealGas(; gas_constant, temperature, gamma)
+Equation of state to describe the relationship between pressure and density
+of a gas using the Ideal Gas Law.
+# Keywords
+- `gas_constant`: Specific gas constant (R) for the gas, typically in J/(kg*K).
+- `temperature` : Absolute temperature of the gas in Kelvin.
+- `gamma`       : Heat-capacity ratio
+This struct calculates the pressure of a gas from its density using the formula:
+\[ P = \rho \cdot R \cdot T \]
+where \( P \) is pressure, \( \rho \) is density, \( R \) is the gas constant, and \( T \) is temperature.
+Note:
+For basic WCSPH this boils down to the assumption of a linear pressure-density relationship.
+"""
+struct StateEquationIdealGas{ELTYPE}
+    gas_constant :: ELTYPE
+    temperature  :: ELTYPE
+    gamma        :: ELTYPE
+
+    function StateEquationIdealGas(; gas_constant, temperature, gamma)
+        new{typeof(gas_constant)}(gas_constant, temperature, gamma)
+    end
+end
+
+function (state_equation::StateEquationIdealGas)(density)
+    (; gas_constant, temperature) = state_equation
+    pressure = density * gas_constant * temperature
+    return pressure
+end
```

Comment:
1. Why is there no background pressure? The SE by Cole is also designed to work with a reference density, at which the pressure is zero. For this state equation to be useful for SPH where we work with relative pressures, it should read `pressure = (density - reference_density) * gas_constant * temperature + background_pressure`.
2. Why does this state equation use the gas constant and doesn't calculate it from the speed of sound like the SE by Cole? This makes it very annoying to switch between state equation, as one has to look up the equation to convert sound speed to gas constant.
3. Please add tests. Codecov reports all of this as uncovered.

Reply:
1. I don't know anymore.
2. c^2/(gamma * T) I think i forgot that i could also do this

Reply:
No I know again why I did that I wanted to avoid setting the physical sound speed.

Reply:
But than I forgot to change it when I decided against doing that.